exp/gateway/validate.go

```diff
@@ -0,0 +1,35 @@
+package gateway
+
+import (
+	"fmt"
+	"math/big"
+	"regexp"
+	"strings"
+)
+
+var (
+	errInvalidHex     = fmt.Errorf("gateway: invalid hex string")
+	errOutOfRangeFelt = fmt.Errorf("gateway: out of range field element")
+)
+
+// isValid returns true if felt is a valid hexadecimal representation of
+// the field element a ∈ 𝔽²ₚ where p = 2²⁵¹ + 17·2¹⁹² + 1 with prefix
+// "0x".
+func isValid(felt string) error {
+	re := regexp.MustCompile(`^0x[a-fA-F0-9]{1,64}$`)
```

Comment:
We can move the regexp compilation to package var declaration to avoid compiling it each time

Reply:
Good catch. Solved.